pkg/proxy/iptables/proxier_test.go

```diff
@@ -770,38 +800,74 @@ func TestNodePort(t *testing.T) {
 	svcName := "svc1"
 	svcIP := net.IPv4(10, 20, 30, 41)
 
-	svc := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: "ns1", Name: svcName}, Port: "80"}
+	svc := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: "ns1", Name: svcName}, Port: "p80"}
 	svcInfo := newFakeServiceInfo(svc, svcIP, 80, api.ProtocolTCP, false)
 	svcInfo.nodePort = 3001
 	fp.serviceMap[svc] = svcInfo
 
-	ep1 := "10.180.0.1:80"
-	fp.endpointsMap[svc] = []*endpointsInfo{{ep1, false}}
+	ip := "10.180.0.1"
+	port := 80
+	fp.allEndpoints = []api.Endpoints{
+		makeTestEndpoints("ns1", svcName, func(ept *api.Endpoints) {
+			ept.Subsets = []api.EndpointSubset{{
+				Addresses: []api.EndpointAddress{{
+					IP: ip,
+				}},
+				Ports: []api.EndpointPort{{
+					Name: "p80",
+					Port: int32(port),
+				}},
+			}}
+		}),
+	}
 
 	fp.syncProxyRules()
 
 	proto := strings.ToLower(string(api.ProtocolTCP))
 	svcChain := string(servicePortChainName(svc, strings.ToLower(proto)))
 
 	kubeNodePortRules := ipt.GetRules(string(kubeNodePortsChain))
-	if !hasJump(kubeNodePortRules, svcChain, "", fmt.Sprintf("%v", svcInfo.nodePort)) {
+	if !hasJump(kubeNodePortRules, svcChain, "", svcInfo.nodePort) {
 		errorf(fmt.Sprintf("Failed to find jump to svc chain %v", svcChain), kubeNodePortRules, t)
 	}
 }
 
+func strPtr(s string) *string {
+	return &s
+}
+
 func TestOnlyLocalLoadBalancing(t *testing.T) {
 	ipt := iptablestest.NewFake()
 	fp := NewFakeProxier(ipt)
 	svcName := "svc1"
 	svcIP := net.IPv4(10, 20, 30, 41)
 
-	svc := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: "ns1", Name: svcName}, Port: "80"}
+	svc := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: "ns1", Name: svcName}, Port: "p80"}
 	svcInfo := newFakeServiceInfo(svc, svcIP, 80, api.ProtocolTCP, true)
 	fp.serviceMap[svc] = typeLoadBalancer(svcInfo)
 
-	nonLocalEp := "10.180.0.1:80"
-	localEp := "10.180.2.1:80"
-	fp.endpointsMap[svc] = []*endpointsInfo{{nonLocalEp, false}, {localEp, true}}
+	ip1 := "10.180.0.1"
+	ip2 := "10.180.2.1"
+	port := 80
+	nonLocalEp := fmt.Sprintf("%s:%d", ip1, port)
+	localEp := fmt.Sprintf("%s:%d", ip2, port)
+	fp.allEndpoints = []api.Endpoints{
+		makeTestEndpoints("ns1", svcName, func(ept *api.Endpoints) {
+			ept.Subsets = []api.EndpointSubset{{
+				Addresses: []api.EndpointAddress{{
+					IP:       ip1,
+					NodeName: strPtr(""),
```

Comment:
er, yeah, I know, I just meant that passing a pointer-to-empty-string seems wrong. The whole point of having the field be a pointer is so that you can pass nil when you don't have a node name to pass, right?